### PR TITLE
deps: use release-polkadot-v1.4.0

### PR DESCRIPTION
**Pull request by Rqnsom**
- Bumping the polkadot deps from polkadot-v1.3.0 to polkadot-v1.4.0
- Updated breaking-change bits

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -13,39 +13,39 @@ repository = "https://github.com/substrate-developer-hub/substrate-node-template
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false, features = ["derive"] }
-scale-info = { default-features = false, features = ["derive"] }
+codec = { package = "parity-scale-codec", version = "3.6", default-features = false, features = ["derive"] }
+scale-info = { version = "2.5", default-features = false, features = ["derive"] }
```
Comment (asmie) [inline runtime/Cargo.toml]: They updated `scale-info` codec to 2.10.0 in the runtime code:
https://github.com/paritytech/polkadot-sdk/blob/fcfdb98ab5c5fce91cce339c8d3e4f5f5844bbe1/substrate/bin/node-template/runtime/Cargo.toml#L17

File: pallets/template/Cargo.toml [hunk 2]
```diff
@@ -13,18 +13,16 @@ repository = "https://github.com/substrate-developer-hub/substrate-node-template
...
 [dependencies]
-codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false, features = [
-	"derive",
-] }
-scale-info = { default-features = false, features = ["derive"] }
-frame-benchmarking = { default-features = false, optional = true, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.3.0" }
-frame-support = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.3.0" }
-frame-system = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.3.0" }
+codec = { package = "parity-scale-codec", version = "3.6", default-features = false, features = ["derive"] }
+scale-info = { version = "2.5", default-features = false, features = ["derive"] }
```
Comment (asmie) [inline pallets/template/Cargo.toml]: Like in our pallet - upstream moved to 2.10.0.

File: runtime/Cargo.toml [hunk 3]
```diff
@@ -13,39 +13,39 @@ repository = "https://github.com/substrate-developer-hub/substrate-node-template
...
+sp-core = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-inherents = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-offchain = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-runtime = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-session = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-std = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-transaction-pool = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
+sp-version = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk.git", branch = "release-polkadot-v1.4.0" }
```
Comment (asmie) [inline runtime/Cargo.toml:43]: Upstream has two more deps here:
https://github.com/paritytech/polkadot-sdk/blob/fcfdb98ab5c5fce91cce339c8d3e4f5f5844bbe1/substrate/bin/node-template/runtime/Cargo.toml#L43C96-L43C96

```
serde_json = { version = "1.0.108", default-features = false, features = ["alloc"] }
sp-genesis-builder = { default-features = false, path = "../../../primitives/genesis-builder" }
```

We don't need them?

**Review by asmie — CHANGES_REQUESTED** [review]
Nice!
I'd stick with the upstream versions of `scale-info` codec.

File: runtime/Cargo.toml (hunk 3 above)
Comment (Rqnsom) [reply runtime/Cargo.toml:43]: Well, it builds without those, but I'll include those so we are aligned with the upstream repo.
Done

File: pallets/template/Cargo.toml (hunk 2 above)
Comment (Rqnsom) [reply pallets/template/Cargo.toml]: Done

File: runtime/Cargo.toml (hunk 1 above)
Comment (Rqnsom) [reply runtime/Cargo.toml]: Done

**Review by asmie — APPROVED** [review]

**Pull request merged**